scripts/helm-nsm-install.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+# A script for wrapping NSM installation via Helm. Both Helm 2 and Helm 3 are
+# supported.
+
+function with_helm_2 () {
+  # We specifically set admission-webhook variables here as it is a subchart
+  # there might be a way to set these as global and refer to them with .Values.global.org
+  # but that seems more intrusive than this hack. Consider changing to global if the charts
+  # get even more complicated
+  helm install --name="$CHART" \
+  --atomic --timeout 300 \
+  --set org="$CONTAINER_REPO",tag="$CONTAINER_TAG" \
+  --set forwardingPlane="$FORWARDING_PLANE" \
+  --set insecure="$INSECURE" \
+  --set global.JaegerTracing="true" \
+  --set spire.enabled="$SPIRE_ENABLED",spire.org="$CONTAINER_REPO",spire.tag="$CONTAINER_TAG" \
+  --set admission-webhook.org="$CONTAINER_REPO",admission-webhook.tag="$CONTAINER_TAG" \
+  --set prefix-service.org="$CONTAINER_REPO",prefix-service.tag="$CONTAINER_TAG" \
+  --namespace="$NSM_NAMESPACE" \
+  deployments/helm/"$CHART"
+}
+
+function with_helm_3 () {
+  # Ensure we have the nsm namespace available
+  echo -n "Ensure namespace $NSM_NAMESPACE exists..."
+  kubectl apply -f k8s/conf/namespace-nsm.yaml &
+  PID=$!
+  sleep 2
+  while [ -d /proc/$PID ]
+  do
+    printf "."
+    sleep 2
+  done
+  echo "Done"
+
+  # We specifically set admission-webhook variables here as it is a subchart
+  # there might be a way to set these as global and refer to them with .Values.global.org
+  # but that seems more intrusive than this hack. Consider changing to global if the charts
+  # get even more complicated
+  echo "Begin chart installation..."
+  helm install "$CHART" \
+  --atomic --timeout 5m \
+  --set org="$CONTAINER_REPO",tag="$CONTAINER_TAG" \
+  --set forwardingPlane="$FORWARDING_PLANE" \
+  --set insecure="$INSECURE" \
+  --set global.JaegerTracing="true" \
+  --set spire.enabled="$SPIRE_ENABLED",spire.org="$CONTAINER_REPO",spire.tag="$CONTAINER_TAG" \
+  --set admission-webhook.org="$CONTAINER_REPO",admission-webhook.tag="$CONTAINER_TAG" \
+  --set prefix-service.org="$CONTAINER_REPO",prefix-service.tag="$CONTAINER_TAG" \
+  --namespace "$NSM_NAMESPACE" \
+  deployments/helm/"$CHART" &
+  sleep 2
+  while [ -d /proc/$PID ]
```

Comment:
That doesn't look like MacOS compatible solution. I'm not sure but I think it is important to us.  Could we consider some solution based on `ps`?

Reply:
I'll look into it, I don't work on a mac so that didn't occur to me. It's a nice touch for folks who develop on macs.